extensions/sql-database-projects/src/models/project.ts

```diff
@@ -20,32 +21,76 @@ import { TelemetryActions, TelemetryReporter, TelemetryViews } from '../common/t
 /**
  * Class representing a Project, and providing functions for operating on it
  */
-export class Project {
-	public projectFilePath: string;
-	public projectFileName: string;
-	public projectGuid: string | undefined;
-	public files: FileProjectEntry[] = [];
-	public dataSources: DataSource[] = [];
-	public importedTargets: string[] = [];
-	public databaseReferences: IDatabaseReferenceProjectEntry[] = [];
-	public sqlCmdVariables: Record<string, string> = {};
-	public preDeployScripts: FileProjectEntry[] = [];
-	public postDeployScripts: FileProjectEntry[] = [];
-	public noneDeployScripts: FileProjectEntry[] = [];
+export class Project implements IProject {
+	private _projectFilePath: string;
```

Comment:
I think it's just something we do in ADS, but it hasn't been done consistently

Reply:
It's something we (partially) inherited from VS Code. We're not super consistent, but I personally prefer it since we aren't always in a place that has support for that stuff (for example Github) and it makes it easy to add a public accessor for it later on if you want to do extra logic.

Reply:
The non-IDE situation is largely resolved by having reasonably-sized functions, though.  Maybe I'm just scarred by Hungarian notation...

Reply:
Hungarian notation is definitely too much and unnecessary IMO

Not sure what you mean by "reasonably sized functions" though. The primary case for this is class-level members which are typically going to be larger than a single page. I would never expect function-level parameters/variables to use _

Reply:
The most common reason I hear for using `_` is so that people know from the name if a variable is local or private, but if the function isn't unnecessarily long, then it's easy to just see if there's a declaration to tell.  That's why I generally think that - while the underscore itself isn't inherently bad - it is sometimes used in lieu of writing cleaner code/as a crutch for sloppier code (not saying that's the case here at all :) ).  Not a red flag, just a minor code smell.

Reply:
Again, I'm confused that you keep saying "function". I've never seen _ used in the context of a function for the reasons you've given above. This is purely about class members where it's pretty likely the declaration won't be visible on the same lines that it's used.

Reply:
Charles and I chatted offline; keep going with the underscores!  My thinking didn't work to differentiate between class-private and class-public variables since everything in TS is lowercased.  Probs should have the linter rule for that enabled if possible.